scripts/install.sh

```diff
@@ -0,0 +1,179 @@
+# |  _ \| '__| |/ _` |/ _` |/ _ \ | __/ _ \  | ' / | | | '_ \ / _ \ '__| '_ \ / _ \ __/ _ \/ __|
+# | |_) | |  | | (_| | (_| |  __/ | || (_) | | . \ |_| | |_) |  __/ |  | | | |  __/ ||  __/\__ \
+# |____/|_|  |_|\__,_|\__, |\___|  \__\___/  |_|\_\__,_|_.__/ \___|_|  |_| |_|\___|\__\___||___/
+#                    |___/
+# usage:
+#    curl -fsSL https://raw.githubusercontent.com/Azure/Bridge-To-Kubernetes/main/scripts/install.sh | bash
+set -e
+set -o pipefail
+set -f
+
+log() {
+    local level=$1
+    shift
+    echo "$(date -u $now) - $level - $*"
+}
+
+# dump uname immediately
+uname -ar
+
+log INFO "Information logged for bridge to kubernetes."
+
+# Try to get os release vars
+# https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html
+# https://stackoverflow.com/questions/394230/how-to-detect-the-os-from-a-bash-script
+if [ -e /etc/os-release ]; then
+    . /etc/os-release
+    DISTRIB_ID=$ID
+else
+    if [ -e /etc/lsb-release ]; then
+        . /etc/lsb-release
+    fi
+fi
+
+if [ -z "${DISTRIB_ID}" ]; then
+    log INFO "Trying to identify using OSTYPE var $OSTYPE "
+    if [[ "$OSTYPE" == "linux"* ]]; then
+        DISTRIB_ID="$OSTYPE"
+        B2KOS="linux"
+        elif [[ "$OSTYPE" == "darwin"* ]]; then
+        DISTRIB_ID="$OSTYPE"
+        B2KOS="osx"
+        elif [[ "$OSTYPE" == "cygwin" ]]; then
+        DISTRIB_ID="$OSTYPE"
+        elif [[ "$OSTYPE" == "msys" ]]; then
+        DISTRIB_ID="$OSTYPE"
+        elif [[ "$OSTYPE" == "win32" ]]; then
+        DISTRIB_ID="$OSTYPE"
+        B2KOS="win"
+        elif [[ "$OSTYPE" == "freebsd"* ]]; then
+        DISTRIB_ID="$OSTYPE"
+    else
+        log ERROR "Unknown DISTRIB_ID or DISTRIB_RELEASE."
+        exit 1
+    fi
+fi
+
+if [ -z "${DISTRIB_ID}" ]; then
+    log ERROR "Unknown DISTRIB_ID or DISTRIB_RELEASE."
+    exit 1
+fi
+
+log INFO "Distro Information as $DISTRIB_ID"
+
+# set distribution specific vars
+PACKAGER=
+SYSTEMD_PATH=/lib/systemd/system
+if [ "$DISTRIB_ID" == "ubuntu" ]; then
+    PACKAGER=apt
+    elif [ "$DISTRIB_ID" == "debian" ]; then
+    PACKAGER=apt
+    elif [[ $DISTRIB_ID == centos* ]] || [ "$DISTRIB_ID" == "rhel" ]; then
+    PACKAGER=yum
+    elif [[ "$DISTRIB_ID" == "darwin"* ]]; then
+    PACKAGER=brew
+    elif [[ "$DISTRIB_ID" == "win32" ]]; then
+    PACKAGER=choco
+else
+    PACKAGER=zypper
+    SYSTEMD_PATH=/usr/lib/systemd/system
+fi
+if [ "$PACKAGER" == "apt" ]; then
+    export DEBIAN_FRONTEND=noninteractive
+fi
+
+# Check JQ Processor and download if not present
+check_jq_processor_present(){
+    log INFO "Checking locally installed JQ Processor version"
+    jqversion=$(jq --version)
+    log INFO "Locally installed JQ Processor version is $jqversion"
+    if [ -z "${jqversion}" ]; then
+        $PACKAGER install jq
+    fi
+}
+
+check_kubectl_present(){
+    log INFO "Checking if kubectl library is present locally"
+    kubectlversion=$(kubectl version --client=true --short)
```

Comment:
>  <img width="899" alt="Screen Shot 2022-10-11 at 8 28 24 AM" src="https://user-images.githubusercontent.com/6233295/194939153-faf649d0-32da-4eba-8899-3cfeddf72788.png">

💡💡 Yeah totally agree there are few ways to do this: 

For `kubectl` we can run `kubectl version --output=json`  **or just `kubectl`

We have options to choose from: (For early release I would say Option-1 is good option and then build on top. Reasons for option-1 is because we should be careful if our installer script changes users PATH (but again depends on the usage and user level of proficiency and comfort)

_Option 1_: (Min support and bail early)

* We check if `kubectl` exist.
* If not, we guide user to get kubectl and that it (Installer fails with helpful message)

_Option 2_: (Full Support)

* We check if `kubectl` exist
* We then `curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.21.0/bin/linux/amd64/kubectl` if user chooses for our installer to install the kubectl.
* Set PATH and move forward.

#### For checking if `command` exist

POSIX standard: https://pubs.opengroup.org/onlinepubs/009696899/utilities/command.html 

```*.sh

if ! [ -x "$(command -v kubectl)" ]; then
  echo 'Error: kubectl is not installed.' >&2
  exit 1
fi

```

Sample only for kubectl command run:

<img width="745" alt="Screen Shot 2022-10-11 at 8 05 29 AM" src="https://user-images.githubusercontent.com/6233295/194936789-5c3f52cf-2746-4a39-af11-6e1de2233701.png">

Reply:
@Tatsinnit I would go with option# 02, I have updated the code in similar way, please let me know for any concerns, Thanks.

Reply:
🎷Cool, sounds good - I think things to ponder will be: `Kubectl version for b2k` 

* https://bridgetokubernetes.blob.core.windows.net/zipv2/lks.json 

Thoughts:

- What is the difference between the `kubectl` shipped from above `zip` vs downloading from main `kubectl` source.
- If the `kubectl shipped with b2k` is special in that case 100 % lets no change user PATH and keep the mechanism.

Reply:
Updated the script to download kubectl, dotnet via packager instead of from mindaro storage account. Both use same url's to download these tools.